backend/hdf5/DimensionHDF5.cpp

```diff
@@ -259,13 +269,156 @@ void SetDimensionHDF5::labels(const none_t t) {
 
 SetDimensionHDF5::~SetDimensionHDF5() {}
 
+//--------------------------------------------------------------
+// Implementation of DataFrameDimensionHDF5
+//--------------------------------------------------------------
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index)
+    : DimensionHDF5(group, index)
+{
+    setType();
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block)
+    : DimensionHDF5(group, index), entity_block(block), entity_file(file)
+{
+    setType();
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame)
+    :DataFrameDimensionHDF5(group, index, file, block)
+{
+    std::shared_ptr<IDataFrame> idf = block->getEntity<IDataFrame>(frame.id());
+    if (!idf)
+        throw std::runtime_error("DataFrameDimensionHDF5 DataFrame not found in block!");
+    if (this->group.hasGroup("data_frame"))
+        this->group.removeGroup("data_frame");
+    auto target = std::dynamic_pointer_cast<DataFrameHDF5>(idf);
+
+    setType();
+    this->group.createLink(target->group(), "data_frame");
+}
+
+DataFrameDimensionHDF5::DataFrameDimensionHDF5(const H5Group &group, ndsize_t index,
+                                               const std::shared_ptr<IFile> &file,
+                                               const std::shared_ptr<IBlock> &block, const DataFrame &frame,
+                                               unsigned col_index)
+    :DataFrameDimensionHDF5(group, index, file, block, frame)
+{
+    this->group.setAttr("column_index", col_index);
+}
+
+boost::optional<unsigned> DataFrameDimensionHDF5::columnIndex() const {
+    unsigned idx;
+    boost::optional<unsigned> col_index;
+
+    if (group.hasAttr("column_index")) {
+        group.getAttr("column_index", idx);
+        col_index = idx;
+    }
+
+    return col_index;
+}
+
+boost::optional<unsigned> DataFrameDimensionHDF5::checkColumnIndex(boost::optional<unsigned> col_index) const {
+    boost::optional<unsigned> column_index = col_index;
+    if (!col_index) {
+        column_index = columnIndex();
+    }
+
+    if (!column_index) {
+        throw nix::OutOfBounds("DataFrameDimensionHDF5: Error accessing column, no column index was given and no default is specified.");
+    }
+
+    nix::DataFrame df = dataFrame();
+    std::vector<Column> cols = df.columns();
```

Comment:
Krass oversight in the `DataFrame` API that we have to fetch all the column information if all we want is the number. Filed as issue #808